File: VB/WebConsole.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.IO
Imports System.Linq
Imports System.Net
Imports System.Text

Namespace SuperImpactor
	Public Class WebConsole
		Public cookie As CookieContainer

		Public Sub New()
			MyBase.New()
		End Sub

		Public Function getKeyValue(ByVal rs As String, ByVal key As String, ByRef value As String, Optional ByVal fromPos As Integer = 0, Optional ByVal keyWithoutQuote As Boolean = False, Optional ByVal quoteChar As String = """") As Object
			Dim obj As Object
			If (Not keyWithoutQuote) Then
				key = String.Concat(quoteChar, key, quoteChar)
			End If
			Dim num As Integer = rs.IndexOf(key, fromPos)
			If (num <> -1) Then
				Dim num1 As Integer = rs.IndexOf(quoteChar, num + key.Length)
				If (num1 <> -1) Then
					num1 = num1 + 1
					Dim num2 As Integer = rs.IndexOf(quoteChar, num1)
					If (num2 <> -1) Then
						value = rs.Substring(num1, num2 - num1)
						obj = ""
					Else
						obj = "incorrect key value 2"
					End If
				Else
					obj = "incorrect key value 1"
				End If
			Else
				obj = "Cannot find key"
			End If
			Return obj
		End Function

		Public Function http(ByVal url As String, Optional ByVal postData As String = "") As String
			Dim str As String
			Try
				Dim httpWebRequest As System.Net.HttpWebRequest = DirectCast(WebRequest.Create(url), System.Net.HttpWebRequest)
				httpWebRequest.AllowAutoRedirect = True
				httpWebRequest.Headers.Add("Accept-Language", "en-us")
				httpWebRequest.Headers.Add("Accept-Encoding", "gzip, deflate")
				httpWebRequest.KeepAlive = True
				httpWebRequest.AutomaticDecompression = DecompressionMethods.GZip Or DecompressionMethods.Deflate
				ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls Or SecurityProtocolType.Tls11 Or SecurityProtocolType.Tls12
				If (Me.cookie Is Nothing) Then
					Me.cookie = New CookieContainer()
				Else
					httpWebRequest.CookieContainer = Me.cookie
				End If
				If (Operators.CompareString(postData, "", False) <> 0) Then
					httpWebRequest.Method = "POST"
					httpWebRequest.GetRequestStream().Write(Encoding.UTF8.GetBytes(postData), 0, Encoding.UTF8.GetBytes(postData).Count())
				End If
				Dim response As HttpWebResponse = DirectCast(httpWebRequest.GetResponse(), HttpWebResponse)
				Dim [end] As String = (New StreamReader(response.GetResponseStream())).ReadToEnd()
				Me.cookie.Add(response.Cookies)
				If (Operators.CompareString(postData, "", False) <> 0) Then
					httpWebRequest.GetRequestStream().Close()
					httpWebRequest.GetRequestStream().Dispose()
				End If
				response.Close()
				response.Dispose()
				str = [end]
			Catch exception As System.Exception
				ProjectData.SetProjectError(exception)
				Interaction.MsgBox(exception.Message, MsgBoxStyle.OkOnly, Nothing)
				str = ""
				ProjectData.ClearProjectError()
			End Try
			Return str
		End Function
	End Class
End Namespace

File: C#/WebConsole.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.IO;
using System.Linq;
using System.Net;
using System.Text;

namespace SuperImpactor
{
	public class WebConsole
	{
		public CookieContainer cookie;

		public WebConsole()
		{
		}

		public object getKeyValue(string rs, string key, ref string value, int fromPos = 0, bool keyWithoutQuote = false, string quoteChar = "\"")
		{
			object obj;
			if (!keyWithoutQuote)
			{
				key = string.Concat(quoteChar, key, quoteChar);
			}
			int num = rs.IndexOf(key, fromPos);
			if (num != -1)
			{
				int num1 = rs.IndexOf(quoteChar, checked(num + key.Length));
				if (num1 != -1)
				{
					num1 = checked(num1 + 1);
					int num2 = rs.IndexOf(quoteChar, num1);
					if (num2 != -1)
					{
						value = rs.Substring(num1, checked(num2 - num1));
						obj = "";
					}
					else
					{
						obj = "incorrect key value 2";
					}
				}
				else
				{
					obj = "incorrect key value 1";
				}
			}
			else
			{
				obj = "Cannot find key";
			}
			return obj;
		}

		public string http(string url, string postData = "")
		{
			string str;
			try
			{
				HttpWebRequest httpWebRequest = (HttpWebRequest)WebRequest.Create(url);
				httpWebRequest.AllowAutoRedirect = true;
				httpWebRequest.Headers.Add("Accept-Language", "en-us");
				httpWebRequest.Headers.Add("Accept-Encoding", "gzip, deflate");
				httpWebRequest.KeepAlive = true;
				httpWebRequest.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
				ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;
				if (this.cookie == null)
				{
					this.cookie = new CookieContainer();
				}
				else
				{
					httpWebRequest.CookieContainer = this.cookie;
				}
				if (Operators.CompareString(postData, "", false) != 0)
				{
					httpWebRequest.Method = "POST";
					httpWebRequest.GetRequestStream().Write(Encoding.UTF8.GetBytes(postData), 0, Encoding.UTF8.GetBytes(postData).Count<byte>());
				}
				HttpWebResponse response = (HttpWebResponse)httpWebRequest.GetResponse();
				string end = (new StreamReader(response.GetResponseStream())).ReadToEnd();
				this.cookie.Add(response.Cookies);
				if (Operators.CompareString(postData, "", false) != 0)
				{
					httpWebRequest.GetRequestStream().Close();
					httpWebRequest.GetRequestStream().Dispose();
				}
				response.Close();
				response.Dispose();
				str = end;
			}
			catch (Exception exception)
			{
				ProjectData.SetProjectError(exception);
				Interaction.MsgBox(exception.Message, MsgBoxStyle.OkOnly, null);
				str = "";
				ProjectData.ClearProjectError();
			}
			return str;
		}
	}
}